pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -0,0 +1,233 @@
+	ClientForMapping func(mapping *meta.RESTMapping) (resource.RESTClient, error)
+
+	builder *resource.Builder
+	mapper  meta.RESTMapper
+	encoder runtime.Encoder
+}
+
+var (
+	selectorLong = dedent.Dedent(`
+		Update the selectors on a resource.
+
+		A selector must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to %[1]d characters.
+		If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.
+                Note: currently selectors can only be set on Service objects.`)
+	selectorExample = dedent.Dedent(`
+                # set the labels and selector before creating a deployment/service pair.
+                kubectl create service cluster-ip -o yaml --dry-run | kubectl set selector --local -f - 'environment=qa' -o yaml | kubectl create -f - 
+                kubectl create deployment my-dep -o yaml --dry-run | kubectl label --local -f - environment=qa -o yaml | kubectl create -f -`)
+)
+
+// NewCmdSelector is the "set selector" command.
+func NewCmdSelector(f cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &SelectorOptions{
+		out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "selector (-f FILENAME | TYPE NAME) EXPRESSIONS [--resource-version=version]",
+		Short:   "Update the selectors on a resource",
+		Long:    fmt.Sprintf(selectorLong),
+		Example: selectorExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args, out))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.RunSelector())
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().Bool("all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().Bool("local", false, "If true, set selector will NOT contact api-server but run locally.")
+	cmd.Flags().String("resource-version", "", "If non-empty, the selectors update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
+	usage := "the resource to update the selectors"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.fnOptions, usage)
+	cmdutil.AddDryRunFlag(cmd)
+	cmdutil.AddRecordFlag(cmd)
+	cmdutil.AddInclude3rdPartyFlags(cmd)
+
+	return cmd
+}
+
+// Complete assigns the SelectorOptions from args.
+func (o *SelectorOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	o.local = cmdutil.GetFlagBool(cmd, "local")
+	o.all = cmdutil.GetFlagBool(cmd, "all")
+	o.record = cmdutil.GetRecordFlag(cmd)
+	o.dryrun = cmdutil.GetDryRunFlag(cmd)
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.changeCause = f.Command()
+	mapper, typer := f.Object()
+	o.mapper = mapper
+	o.encoder = f.JSONEncoder()
+
+	o.builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
```

Comment:
You can now use the factory to create the resource builder for you. Try f.NewBuilder()